src/workerd/io/actor-sqlite.c++

```diff
@@ -533,39 +537,34 @@ kj::Maybe<kj::Promise<void>> ActorSqlite::put(kj::Array<KeyValuePair> pairs, Wri
   requireNotBroken();
   // TODO(cleanup): Most of this code comes from DurableObjectStorage::transactionSync and could be re-used.
   if (currentTxn.is<NoTxn>()) {
-    // If we are not in a transaction, let's use an ExplicitTxn, which should rollback automatically
-    // if some put fails.
+    // If we are not in a transaction, start an ImplicitTxn since that's what would happen on the
+    // first write anyway.
+    startImplicitTxn();
+  }
 
-    // TODO(someday) this should really start an ImplicitTxn, which has the advantage of
-    // supporting allowUnconfirmed.
-    disableAllowUnconfirmed(options, "multi put will create an ExplicitTxn");
+  KJ_ASSERT(!currentTxn.is<NoTxn>());
 
-    auto txn = startTransaction();
-    txn->put(kj::mv(pairs), options);
-    txn->commit();
```

Comment:
Oof, this previous behavior is wrong:
* It violates our atomiticy guarantees. But this will only be a real problem with local workerd; in production we are saved by SRS later on.
* If the app does several put()s in a row, it'll force a separate transaction for each one, which is much less efficient than a single merged transaction. So this may have been a performance regression since it rolled out.

This PR seems to be fixing the problems, so that's great.